Core/Base/ElasticDB.py

```diff
@@ -0,0 +1,96 @@
+########################################################################
+# $Id: $
+########################################################################
+
+""" ElasticDB is a base class used to connect an Elasticsearch database and manages 
+queries.
+"""
+
+__RCSID__ = "$Id$"
+
+from DIRAC                                       import gLogger, gConfig
+from DIRAC.Core.Utilities.ElasticSearchDB        import ElasticSearchDB
+from DIRAC.ConfigurationSystem.Client.Utilities  import getElasticDBParameters
+
+class ElasticDB( ElasticSearchDB ):
+
+  """
+  .. class:: ElasticDB
+
+  :param str dbHost: the host name of the Elasticsearch database
+  :param str dbPort: The port where the Elasticsearch database is listening
+  :param str clusterName: The name of the cluster.
+  """
+  __dbHost = None
+  __dbPort = None
+  __clusterName = ""
+  
+  ########################################################################
+  def __init__( self, dbname, fullName, debug = False ):
+    """ c'tor
+
+    :param self: self reference
+    :param str dbName: name of the database for example: MonitoringDB
+    :param str fullName: The full name of the database for example: 'Monitoring/MonitoringDB'
+    :param bool debug: save the debug information to a file 
+    """
+    
+    database_name = dbname
+    self.log = gLogger.getSubLogger( database_name )
+
+    result = getElasticDBParameters( fullName )
+    if not result['OK'] :
+      raise RuntimeError( 'Cannot get database parameters: %s' % result['Message'] )
+
+    dbParameters = result[ 'Value' ]
+    self.__dbHost = dbParameters[ 'Host' ]
+    self.__dbPort = dbParameters[ 'Port' ]
+    
+    super( ElasticDB, self ).__init__( self.__dbHost, self.__dbPort, debug = debug )
+
+    if not self._connected:
+      raise RuntimeError( 'Can not connect to DB %s, exiting...' % self.dbName )
+
+
+    self.log.info( "==================================================" )
+    self.log.info( "Host:           " + self.__dbHost )
+    self.log.info( "Port:           " + str( self.__dbPort ) )
+    self.log.info( "ClusterName:    " + self.__clusterName )
+    self.log.info( "==================================================" )
+  
+  ########################################################################
+  def setDbHost( self, hostName ):
```

Comment:
again it is depend on the developer. The developers who have experience in scripting languages do not see the advantages of the getter and setter methods. It is useless now, because the values are not modified. But if later on someone implement a mew class and wants to modify the value the recommended way is to use setter method. I know if you want to see who modify the value you can search for the name. But when you have to debug? If the variable will be modified 10 times and the person is not familiar the code, he/she has to write 10 times a print message in order to see which line modified the value. But if we have a setter method, one print with the value and stack trace will simplify everything....